logstash-core/lib/logstash/api/modules/settings.rb

```diff
@@ -0,0 +1,36 @@
+# encoding: utf-8
+module LogStash
+  module Api
+    module Modules
+      class Settings < ::LogStash::Api::Modules::Base
+        def handle_logging(logging_keys, settings)
+          logging_keys.each do |key|
+            level = settings.delete(key)
+            path = key.split("logger.")[1]
+            if level.nil?
+              raise ArgumentError, "[log.level] must be specified"
+            else
+              LogStash::Logging::Logger::configure_logging(level, path)
+            end
+          end
+        end
+
+        put "/" do
+          # only logging setting for now
+          begin
+            req_body = LogStash::Json.load(request.body.read)
+            handle_logging(req_body.select { |k,_| k.start_with?("logger.") }.keys, req_body)
+            unless req_body.empty?
+              raise ArgumentError, "unrecognized option [#{req_body.keys.first}]"
```

Comment:
I wonder, can we do this inside `handle_logging` instead? It feels like an odd side effect to check for success by having `handle_logging` modify the given `req_body` and checking if the modifications leave the value empty afterwards.